File: Server/src/model/buff/towerbuff/TowerBuff.java
package model.buff.towerbuff;

import model.Define.ModelDefine;
import model.Tower.Tower;
import model.buff.Buff;
import model.buff.Effect;
import util.readJson.ModelJson;
import util.readJson.ReadJson;

import java.util.LinkedList;

public class TowerBuff extends Buff {
    protected int towerBuffType;
    protected String name;
    protected String durationType;
    public TowerBuff(int towerBuffType,int numEffect) {
        super(ModelDefine.BUFF_TOWERBUFF,towerBuffType);
        this.towerBuffType = towerBuffType;
        this.readTowerBuffJson(numEffect);
    }
    public void readTowerBuffJson(int numEffect){
        ReadJson towerBuffJson = new ReadJson(ModelJson.TowerBuff_Json);
        String pointer = towerBuffJson.get();
        pointer = towerBuffJson.getWithJsonString(pointer,"towerBuff");
        pointer = towerBuffJson.getWithJsonString(pointer,String.valueOf(this.towerBuffType));
        this.name = towerBuffJson.getWithJsonString(pointer,"name");
        this.durationType = towerBuffJson.getWithJsonString(pointer,"durationType");
        pointer = towerBuffJson.getWithJsonString(pointer,"effects");
        for(int i = 1; i <= numEffect; i++){
            String source = towerBuffJson.getWithJsonString(pointer,String.valueOf(i));
            System.out.println(source);
            source = towerBuffJson.removeOneBraket(source);
            System.out.println(source);
            this.effectList.add(new Effect(source)) ;
        }
    };
    @Override
    public void updateBuff(double dt) {

    }

}

File: Client/src/modules/models/buff/TowerBuff.js
var TowerBuff = Buff.extend({
    ctor: function (towerBuffType,numEffect = 3) {
        this._super(GC.TOWERBUFF.ID,towerBuffType);
        this.towerBuffType = towerBuffType;
        ConfigLoader.load(this, GC.TOWERBUFF.LIST[towerBuffType]);
        this.readTowerBuff(numEffect);
    },
    readTowerBuff: function (numEffect){
        for(let i = 1; i <= numEffect; i++){
            this.name = "";
            this.type = "";
            this.value = 0;
            ConfigLoader.load(this, this.effects["" + i]);
            //cc.log("TowerBuff "+this.name + " " +this.type + " "+this.value);
            //cc.log("TowerBuff "+this.durationType);
            //cc.log("TowerBuff "+this.effects[i] + " 99 " + this.effects[i]);
            let effect = new Effect(this.name,this.type,this.value)
            this.addEffect(effect);
        }
    }
})